tests/test_tokenization_xlm_roberta.py

```diff
@@ -89,9 +248,11 @@ def test_tokenization_base_hard_symbols(self):
             1098,
             29367,
             47,
-            4426,
-            3678,
-            2740,
+            # 4426, # What fairseq tokenizes from "<unk>": "_<"
+            # 3678, # What fairseq tokenizes from "<unk>": "unk"
+            # 2740, # What fairseq tokenizes from "<unk>": ">"
+            3,  # What we tokenize from "<unk>": "<unk>"
+            6,  # Residue from the tokenization: an extra sentencepiece underline
```

Comment:
Concerning this, I think we have no choice but to have a discrepancy with the fairseq implementation. The original implementation does not handle tokenizing the unknown token to an ID, "<unk>" will get tokenized to `("_<", "unk", ">")`.

Since we require our tokenizers to do the following:

```py
ids = tokenizer.encode(tokenizer.decode(ids))
```
and
```py
sequence = tokenizer.decode(tokenizer.encode(sequence))
```
We have to handle the conversion from the unknown token to its ID.

*However*, there is a residue from this transformation which is the rogue SentencePiece underline. I don't know what's the best option to manage this, or if it should be managed at all.

Reply:
I see. XLM-R is designed to work on raw text so I guess we should not meet the `<unk>` token in normal condition and there is probably no real need to go too far on this indeed.